discord_google_calendar_bot.py

```diff
@@ -0,0 +1,338 @@
+        "description": discord_event.description,
+        "start": {
+            "dateTime": discord_event.start_time.isoformat(),
+            "timeZone": "UTC",
+        },
+        "end": {
+            "dateTime": end_time.isoformat(),
+            "timeZone": "UTC",
+        },
+    }
+
+    # Create the calendar event based on Discord event ID and announce it to
+    # the console
+    try:
+        google_client.events().insert(
+            calendarId=CALENDAR_ID, body=google_event_details
+        ).execute()
+    except HttpError as error:
+        if error.status_code == 409:
+            update_google_event(discord_event, discord_event)
+    else:
+        print(f"Google Calendar event created for {discord_event.name}")
+
+
+def update_google_event(old_discord_event, new_discord_event):
+    """
+    Updates a Google Calendar event based on changes in a Discord event.
+
+    Args:
+        old_discord_event (discord.Event): The previous version of the
+        Discord event.
+        new_discord_event (discord.Event): The updated version of the
+        Discord event.
+
+    This function takes the old and new versions of a Discord event and
+    updates the corresponding Google Calendar event. It generates the event
+    details, including the summary, location, description, start time,
+    and end time, based on the information in the new Discord event.
+
+    The function then uses the Google Calendar API to update the existing
+    event in the calendar, using the event ID from the old Discord event.
+    Finally, it prints a message to the console indicating that the Google
+    Calendar event has been updated.
+    """
+    google_event_summary_prefix = "Discord (" + discord_client.guilds[0].name + "): "
+
+    if new_discord_event.entity_type is discord.EntityType.voice:
+        discord_end_time = new_discord_event.start_time
+        discord_location = (
+            GOOGLE_EVENT_CHANNEL_NAME_PREFIX + new_discord_event.channel.name
+        )
+    else:
+        discord_end_time = new_discord_event.end_time
+        discord_location = GOOGLE_EVENT_LOCATION_PREFIX + new_discord_event.location
+
+    google_event_details = {
+        "id": str(new_discord_event.id),
+        "summary": google_event_summary_prefix + new_discord_event.name,
+        "location": discord_location,
+        "description": new_discord_event.description,
+        "start": {
+            "dateTime": new_discord_event.start_time.isoformat(),
+            "timeZone": "UTC",
+        },
+        "end": {
+            "dateTime": discord_end_time.isoformat(),
+            "timeZone": "UTC",
+        },
+    }
+
+    # Update the calendar event based on Discord event ID and announce it to
+    # the console
+    google_client.events().update(
+        calendarId=CALENDAR_ID,
+        eventId=str(old_discord_event.id),
+        body=google_event_details,
+    ).execute()
+    print(f"Google Calendar event updated for {new_discord_event.name}")
```

Comment:
instead of using print statements, use logger from the logging standard library. Once you do that, use % lazy strings, and not f-strings.

Reply:
I replaced the print statements with logging to stdout at an "INFO" level. This is compatible with running it manually, as a systemd service, and a Docker container.